### PR TITLE
Refactor `gulpfile.js`

### PR DESCRIPTION
**Pull request by Raruto**
a partial refactor is actually needed (or at least suggested) before complete [#68](https://github.com/g3w-suite/g3w-client/issues/68)

based on https://github.com/g3w-suite/g3w-client/pull/35

---

**Notable changes:**

- remove function `interpolateVersion()` 
- remove function `set_current_hash_version()`
- remove task `g3w-admin:client_only`
- remove task `set_build_all_to_false`
- remove variable `g3w_admin_version`

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -1,58 +1,70 @@
...
+
+const packageJSON = require('./package.json');
+const g3w         = require('./config');
+
+///////////////////////////////////////////////////////
+
+/**
+ * TODO: can we safely delete the followings ?
+ */
+
```
Comment (Raruto) [inline gulpfile.js:51]: @volterra79 can you check this before merge?

File: gulpfile.js [hunk 2]
```diff
@@ -71,422 +83,496 @@ const buildChanges = {
...
+gulp.task('g3w-admin-plugins-select',            ['g3w-admin:plugins']);
+gulp.task('g3w-admin-client:static',             ['g3w-admin:static']);
+gulp.task('g3w-admin-client:template',           ['g3w-admin:templates']);
+
+//////////////////////////////////////////////////////////////////////////
+/**
+ * TODO: check which of the following tasks is actually required
+ */
+//////////////////////////////////////////////////////////////////////////
+
```
Comment (Raruto) [inline gulpfile.js:544]: @volterra79 can you check this before merge?

Comment (volterra79) [reply gulpfile.js:544]: ok .. i let you know as soon as possible

**Comment by Raruto** [comment]
@volterra79 if that's okay with you I'd say we can quite safely merge this.

**Review by volterra79 — APPROVED** [review]

**Pull request merged**